pygfx/utils/text/_fontfinder.py

```diff
@@ -0,0 +1,265 @@
+"""
+Detect fonts on the system.
+Some of this code was "inspired" by the Matplotlib font manager.
+"""
+
+import os
+import sys
+import json
+import time
+import secrets
+
+from .. import logger, get_resources_dir, get_cache_dir
+
+
+def find_fonts_from_dir(directory, recursive):
+    """
+    Return a list of all fonts matching any of the extensions, found
+    recursively under the directory.
+    """
+    if not os.path.isdir(directory):
+        raise OSError(f"Not a directory: {directory}")
+    extensions = ".ttf", ".otf"
+    dir_paths = set()
+    file_paths = set()
+    if not recursive:
+        dir_paths.add(directory)
+        for fname in os.listdir(directory):
+            if fname.lower().endswith(extensions):
+                file_paths.add(os.path.join(directory, fname))
+    else:
+        for dirpath, _, filenames in os.walk(directory):
+            dir_paths.add(dirpath)
+            for fname in filenames:
+                if fname.lower().endswith(extensions):
+                    file_paths.add(os.path.join(dirpath, fname))
+    return dir_paths, file_paths
+
+
+def get_builtin_fonts():
+    """Get a list of fonts that is shipped with PyGfx."""
+    dir_paths, file_paths = find_fonts_from_dir(get_resources_dir(), False)
+    return file_paths
+
+
+def get_entrypoints_fonts():
+    """Get a list of fonts defined via the setuptools entrypoint. TODO someday."""
+    raise NotImplementedError()
```

Comment:
I think we need to add one more directory fonts can be found in, and use an environment variable to configure it's location. That will allow users to ship font files with applications. Should be trivial.

Reply:
We have `font_manager.add_font_file()`.